src/app/forms/services/dynamic-form.service.ts

```diff
@@ -123,32 +124,31 @@ export class DynamicFormService {
     validators.forEach(v => control.addAsyncValidators(this.asyncValidatorMap[v.name](v.args)));
   }
 
-  static updateValidity(form: FormGroup | FormArray, errors: GlobalError[]) {
+  static updateValidity(form: FormGroup | FormArray | CustomFormGroup | CustomFormArray, errors: GlobalError[]) {
```

Comment:
All form components use this code. What was the reason for the change? And was this tested?

Reply:
I suggested this change, the reason behind it is currently this function accepts FormGoups and FormArrays. The issue is that there are some places where it accesses `.meta` of those classes. The issue with doing that is that Formgroups and FormArrays don't have `.meta` as this is only for `CustomFormGroups` and `CustomFormArrays`. Basically I was getting errors on my branch where it was reading `.changeDetection` of undefined, I believe. And Owen was running into some similar issues, although it seems they were actually do to something else. 

To be honest, now looking at the code, I don't know if the issue I was having was actually do to this code. What I can say is that this function was recently changed where we did tech record create in order to accept `FormGroup | FormArray` as an input, whereas it used to be `CustomFormGroup | CustomFormArray`.

The change effectively allows you to run `updateValidity` on both regular angular `FormGroup | FormArray | FormControl | AbstractControl` and `CustomFormGroups | CustomFormArray | CustomFormControl`, whilst preserving type safety.

I'm not too attached to the change to be honest, but it's realistically quite a small one.

Reply:
I've been looking at it too and tried it out this morning, I think in actual fact the issue was not related to this code, and was actually an issue with one of the subsiquent changes. So seen as this change is doing nothing for this ticket I'll revert the changes and just keep a note of them seen as we're thinking it wasn't actuall this code for both of our issues.

Reply:
Thanks for the explanation @gjulien-bjss. Core functionality should be altered based on valid reasons. If your bug was not related to this, then thanks for reverting @owen-corrigan-bjss. Otherwise, this could have become another loose end if and when something broke with dynamic forms, making us think _"was it that last change we made that broke this"_.